lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -532,12 +558,20 @@ bool DiagnosticVerifier::verifyFile(unsigned BufferID,
     }
 
     if (I == CapturedDiagnostics.end()) continue;
-    
-    auto StartLoc = SMLoc::getFromPointer(expected.MessageRange.begin());
-    auto EndLoc = SMLoc::getFromPointer(expected.MessageRange.end());
-    
-    llvm::SMFixIt fixIt(llvm::SMRange{ StartLoc, EndLoc }, I->getMessage());
-    addError(expected.MessageRange.begin(), "incorrect message found", fixIt);
+
+    if (I->getMessage().find(expected.MessageStr) == StringRef::npos) {
+      auto StartLoc = SMLoc::getFromPointer(expected.MessageRange.begin());
+      auto EndLoc = SMLoc::getFromPointer(expected.MessageRange.end());
+
+      llvm::SMFixIt fixIt(llvm::SMRange{StartLoc, EndLoc}, I->getMessage());
+      addError(expected.MessageRange.begin(), "incorrect message found", fixIt);
+    } else {
+      // The difference must be only in the column
```

Comment:
Is this something we can assert?

Reply:
yeah, good idea! I made this an if-elseif and made the last branch unreachable